mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1035,6 +1036,50 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			})
 		}
 	})
+
+	// These tests only run when a KMS mock server is running on localhost:8000.
+	mt.RunOpts("kms tls tests", noClientOpts, func(mt *mtest.T) {
+		testcases := []struct {
+			name       string
+			envValue   string
+			errMessage string
+		}{
+			{
+				"invalid certificate",
+				"INVALID_CERT",
+				"expired",
+			},
+			{
+				"invalid hostname",
+				"INVALID_HOSTNAME",
+				"SANs",
+			},
+		}
+
+		for _, tc := range testcases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				// Only run test if correct KMS mock server is running.
+				if os.Getenv("KMS_TLS_TESTCASE") == tc.envValue {
```

Comment:
nit: We should aim to get all of the env var checking and `Skip` calls out of way as early as possible. I'd recommend refactoring this to:
```
if os.Getenv(...) != tc.envValue {
    mt.Skipf(...)
}

// regular test logic
```
This makes the skipping conditions and message obvious at the very beginning of the test and avoids the extra level of indentation for the main test logic.

Reply:
Sounds good! Added check to the beginning.